bionic/deriver.py

```diff
@@ -301,28 +304,28 @@ def register_provider(provider):
         # behavior downstream.
         self._followup_dnode_lists_by_dnode = dict(followup_dnode_lists_by_dnode)
 
-    def _set_up_bootstrap(self):
+    def _set_up_core(self):
         """
         Initializes some key objects needed to compute user-defined entities.
         """
 
-        if self._bootstrap is not None:
+        if self._core_is_finalized:
```

Comment:
If we are using `finalize` to describe this variable, should we also use `finalize` for this method instead of `set_up`?

Also, since we have a default core, should this be a property of `ExecutionCore` instead?

Reply:
I agree, we can improve this name. What do you think about `set_up_final_core`?

I'm not so sure about making `is_finalized` a property of `ExecutionCore` though. I originally had it this way, but I moved it out. My thinking is, we need to put this boolean somewhere, but putting it in `EntityDeriver` hides it from the other classes that use `ExecutionCore`. In particular, the fact that the deriver has two stages and switches cores doesn't seem like a detail that needs to be exposed to the logic in `flow_execution` and `task_execution`.

@simonafk For the same reason, having an `initialize` method that mutates the core doesn't seem very attractive to me. Currently the core is mutable from the POV of the `EntityDeriver` but immutable from the POV of the execution code; with this change it would be mutable everywhere. WDYT?

Reply:
`set_up_final_core` reads cleaner to me than just `set_up_core`. But in that case, wdyt about renaming this variable `_final_core_is_set_up`? I think that if that doesn't sound clean either, then maybe we aren't using the right verbs.

Reply:
How about `_core_is_final`?